src/pages/upload/index.tsx

```diff
@@ -122,6 +131,14 @@ const Upload: React.FC<FormikProps<FormProps> & {instructors: Instructor[]}> = (
   const [courseInstructorId, setCourseInstructorId] = React.useState<string>(
     viewer?.instructor_id || instructors[0]?.eggheadInstructorId,
   )
+  const sanityCollaboratorId: string | undefined = _find(instructors, [
+    'eggheadInstructorId',
+    courseInstructorId,
+  ])?._id
+
+  React.useEffect(() => {
+    setFieldValue('course.collaboratorId', sanityCollaboratorId)
+  }, [sanityCollaboratorId, setFieldValue])
```

Comment:
Match the instructor selected from the drop down to their `_id` and put that into Formik's values that will be submitted with the rest of the course builder metadata.